src/runner.rs

```diff
@@ -46,79 +94,145 @@ pub async fn run_test_command(
+                Err(e) => {
+                    log::warn!("Error opening file: {}", e);
+                    continue;
+                }
+            };
+            let reader = std::io::BufReader::new(file);
+            let junitxml = match junit_parser::from_reader(reader) {
+                Ok(junitxml) => junitxml,
+                Err(e) => {
+                    log::warn!("Error parsing junitxml: {}", e);
+                    continue;
+                }
+            };
+            for suite in junitxml.suites {
+                let parent_name = suite.name;
+                for case in suite.cases {
+                    let failure = case.status.is_failure();
+                    if failure {
+                        let name = case.original_name;
+                        log::debug!("Test failed: {} -> {}", parent_name, name);
+                        failures.push(Test {
+                            parent_name: parent_name.clone(),
+                            name: name.clone(),
+                            class_name: case.classname.clone(),
+                            file: case.file.clone(),
+                        });
                     }
                 }
             }
         }
     }
-    let exit_code = result.code().unwrap_or(EXIT_FAILURE);
-    Ok(RunResult {
+    Ok(failures)
+}
+
+pub async fn run_quarantine(
+    run_result: &RunResult,
+    api_address: &str,
+    token: &str,
+    org_url_slug: &str,
+    repo: &BundleRepo,
+    delay: std::iter::Take<ExponentialBackoff>,
+    use_quarantining: bool,
+) -> anyhow::Result<QuarantineRunResult> {
+    if !use_quarantining {
+        log::info!("Skipping quarantining step.");
+        return Ok(QuarantineRunResult {
+            exit_code: run_result.exit_code,
+            quarantine_status: QuarantineBulkTestStatus {
+                group_is_quarantined: false,
+                quarantine_results: Vec::new(),
+            },
+        });
+    }
+    // check with the API if the group is quarantined
+    log::debug!(
+        "Checking quarantine status for failures: {:?}",
+        run_result.failures
+    );
+    let quarantine_results = if run_result.failures.is_empty() {
+        QuarantineBulkTestStatus {
+            group_is_quarantined: false,
+            quarantine_results: Vec::new(),
+        }
+    } else {
+        let result = Retry::spawn(delay, || {
+            get_quarantine_bulk_test_status(
+                api_address,
+                token,
+                org_url_slug,
+                &repo.repo,
+                &run_result.failures,
+            )
+        })
+        .await;
+
+        result.unwrap_or_else(|e| {
+            log::error!("Failed to get quarantine results: {:?}", e);
+            QuarantineBulkTestStatus {
+                group_is_quarantined: false,
+                quarantine_results: Vec::new(),
+            }
+        })
+    };
+    log::info!("Quarantine results: {:?}", quarantine_results);
```

Comment:
I think this should also be either:
1) Printed one result per line.
2) Moved to debug.
3) Controlled by some kind of dry-run flag.

To me it seems like printing a structure that may contain a lot of entries using a single line sounds like a bad practice. It's not useful / too verbose and it's hard to read.

Reply:
Here is how we handle similar case in `main.rs`. There is a flag to print all XML files:
```rust
    if print_files {
        println!("Files to upload:");
        for file_set in &meta.file_sets {
            println!(
                "  File set ({:?}): {}",
                file_set.file_set_type, file_set.glob
            );
            for file in &file_set.files {
                println!("    {}", file.original_path);
            }
        }
    }
```